eksrollup/lib/k8s.py

```diff
@@ -152,6 +152,131 @@ def taint_node(node_name):
         logger.info("Exception when calling CoreV1Api->patch_node: {}".format(e))
 
 
+def deployment_restart(node_name):
+    """
+    Availability is not guranteed for a single replica deployment.
+    PDB only works with more than 1 replica. Node draining without
+    PDB is synchronous, meaning new pods are started at the same
+    time old pods are being terminated. A deployment restart will
+    ensure there is always one pod available to the service. If
+    enabled, this function will only perform a rolling restart on
+    deployments with single replicas.
+    """
+    # Set threads for the kubectl command
+    threads = []
+
+    ensure_config_loaded()
+
+    k8s_api = client.CoreV1Api()
+
+    apps_k8s_api = client.AppsV1Api()
+
+    # Node field selector used to discover all pods on the node
+    node_field_selector = "spec.nodeName={}".format(node_name)
```

Comment:
What Python version is this using? If 3.6 or above, then just use f-strings:

```python
node_field_selector = f"spec.nodeName={node_name}"
```

Reply:
Yeah, this is using python 3.6+. Updated. Thanks!